backend/fleet_management/tests/test_refuels.py

```diff
@@ -0,0 +1,73 @@
+from django.contrib.auth.models import Group
+
+from djmoney.money import Money
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from fleet_management.constants import Groups
+from fleet_management.factories import UserFactory, RefuelFactory, CarFactory
+from fleet_management.models import Refuel
+
+
+class RefuelsApiTestCase(APITestCase):
+    def setUp(self):
+        self.url = reverse("refuels")
+        self.user = UserFactory.create(
+            groups=[Group.objects.get(name=Groups.Driver.name)]
+        )
+        self.refuels = RefuelFactory.create_batch(size=3)
+
+        self.car = CarFactory.create()
+        self.driver = UserFactory.create()
+        self.currency = Money(25, "USD")
+
+    def test_401_for_unlogged_user(self):
+        res = self.client.get(self.url)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_get_all_refuels(self):
+        self.client.force_login(self.user)
+        res = self.client.get(self.url)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(self.refuels), len(res.data))
+        for idx, refuel in enumerate(sorted(res.data, key=lambda p: p["id"])):
+            self.assertEqual(refuel["car"], res.data[idx]["car"])
+            self.assertEqual(refuel["date"], res.data[idx]["date"])
+            self.assertEqual(
+                refuel["current_mileage"], res.data[idx]["current_mileage"]
+            )
+            self.assertEqual(
+                refuel["refueled_liters"], res.data[idx]["refueled_liters"]
+            )
+            self.assertEqual(
+                refuel["price_per_liter"], res.data[idx]["price_per_liter"]
+            )
+            self.assertEqual(refuel["currency"], res.data[idx]["currency"])
+
+    def test_create_a_refuel(self):
+        payload = {
+            "car": {"id": self.car.id},
+            "driver": {"id": self.driver.id},
+            "date": "2020-09-25",
+            "current_mileage": 0,
+            "refueled_liters": 5,
+            "price_per_liter": 34,
+            "currency.amount": str(self.currency.amount),
+            "currency.currency": self.currency.currency.code,
+        }
+
+        self.client.force_login(self.user)
+        res = self.client.post(self.url, data=payload, format="json")
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+
+        refuel = Refuel.objects.filter(id=res.data["id"])
+        self.assertEqual(refuel.count(), 1)
+
+        refuel = refuel[0]
```

Comment:
```suggestion
        refuel = Refuel.objects.get(id=res.data["id"])
        self.assertEqual(refuel.count(), 1)

```
why not that? you're fetching the record by `id` anyway, so it's ensured by DB that you won't find two records with same id. Also, the `refuel = refuel[0]` won't be needed then.

Reply:
another bump. I don't want to 'force' my changes but I'd like to hear your reasoning behind it.

Reply:
I was looking at convention of previous tests and in drives we have:

```        self.assertEqual(drive.count(), 1)

        drive = drive[0]
        self.assertTrue(drive.is_verified)```